src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php

```diff
@@ -22,108 +23,161 @@ class AblyBroadcaster extends Broadcaster
+    {
+        $channelName = $request->channel_name;
+        $token = $request->token;
+        $connectionId = $request->socket_id;
+        $normalizedChannelName = $this->normalizeChannelName($channelName);
+        $userId = null;
+        $channelCapability = ['*'];
+        $user = $this->retrieveUser($request, $normalizedChannelName);
+        if ($user) {
+            $userId = method_exists($user, 'getAuthIdentifierForBroadcasting')
+                ? $user->getAuthIdentifierForBroadcasting()
+                : $user->getAuthIdentifier();
+        }
+        if ($this->isGuardedChannel($channelName)) {
+            if (! $user) {
+                throw new AccessDeniedHttpException('User not authenticated, '.$this->stringify($channelName, $connectionId));
+            }
+            try {
+                $userChannelMetaData = parent::verifyUserCanAccessChannel($request, $normalizedChannelName);
+                if (is_array($userChannelMetaData) && array_key_exists('capability', $userChannelMetaData)) {
+                    $channelCapability = $userChannelMetaData['capability'];
+                    unset($userChannelMetaData['capability']);
+                }
+            } catch (\Exception $e) {
+                throw new AccessDeniedHttpException('Access denied, '.$this->stringify($channelName, $connectionId, $userId), $e);
+            }
+        }
 
-        $broadcastIdentifier = method_exists($user, 'getAuthIdentifierForBroadcasting')
-                    ? $user->getAuthIdentifierForBroadcasting()
-                    : $user->getAuthIdentifier();
+        try {
+            $signedToken = $this->getSignedToken($channelName, $token, $userId, $channelCapability);
+        } catch (\Exception $_) { // excluding exception to avoid exposing private key
+            throw new AccessDeniedHttpException('malformed token, '.$this->stringify($channelName, $connectionId, $userId));
+        }
 
-        $signature = $this->generateAblySignature(
-            $request->channel_name,
-            $request->socket_id,
-            $userData = array_filter([
-                'user_id' => (string) $broadcastIdentifier,
-                'user_info' => $result,
-            ])
-        );
+        $response = ['token' => $signedToken];
+        if (isset($userChannelMetaData) && is_array($userChannelMetaData) && count($userChannelMetaData) > 0) {
+            $response['info'] = $userChannelMetaData;
+        }
 
-        return [
-            'auth' => $this->getPublicToken().':'.$signature,
-            'channel_data' => json_encode($userData),
-        ];
+        return $response;
     }
 
     /**
-     * Generate the signature needed for Ably authentication headers.
+     * Return the valid authentication response.
      *
-     * @param  string  $channelName
-     * @param  string  $socketId
-     * @param  array|null  $userData
-     * @return string
+     * @param $request \Illuminate\Http\Request
+     * @param $result mixed
```

Comment:
```suggestion
     * @param  \Illuminate\Http\Request  $request
     * @param  mixed  $result
```
First variable type and then the variable name

Reply:
Fixed!